### ISSUE TITLE
Contextual glyph replacements – would require some new and some changed graphics

### ISSUE THREAD

**Issue by Crissov** [Ranks/emojione#393]
There are some standardized emoji sequences using U+200D Zero-Width Joiner (ZWJ) which shall lead to a new combined glyph – a ligature in families, a new graphic in professions –, but there are also some sequences that are actually used in the wild where either or both of the glyphs should take a special shape. I’m giving but a few examples for now:

![🚂 U+1F682](https://raw.githubusercontent.com/Ranks/emojione/master/assets/png/1f682.png)![🚃 U+1F683](https://raw.githubusercontent.com/Ranks/emojione/master/assets/png/1f683.png) ![🚋 U+1F68B](https://raw.githubusercontent.com/Ranks/emojione/master/assets/png/1f68b.png) ![🚞 U+1F69E](https://raw.githubusercontent.com/Ranks/emojione/master/assets/png/1f69e.png) ![🚅 U+1F685](https://raw.githubusercontent.com/Ranks/emojione/master/assets/png/1f685.png) ![🚈 U+1F688](https://raw.githubusercontent.com/Ranks/emojione/master/assets/png/1f688.png) ![🚝 U+1F69D](https://raw.githubusercontent.com/Ranks/emojione/master/assets/png/1f69d.png) ![🚟 U+1F69F](https://raw.githubusercontent.com/Ranks/emojione/master/assets/png/1f69f.png) ![🚄 U+1F684](https://raw.githubusercontent.com/Ranks/emojione/master/assets/png/1f684.png) ![🚆 U+1F686](https://raw.githubusercontent.com/Ranks/emojione/master/assets/png/1f686.png) ![🚇 U+1F687](https://raw.githubusercontent.com/Ranks/emojione/master/assets/png/1f687.png) ![🚊 U+1F68A](https://raw.githubusercontent.com/Ranks/emojione/master/assets/png/1f68a.png) ![🛲 U+1F6F2](https://raw.githubusercontent.com/Ranks/emojione/master/assets/png/1f6f2.png)

- 🚂🚃 – the Railway Car works well enough with the Steam Locomotive in Emojione, at least at small sizes
- 🚋🚃, 🚞🚃 – it kinda works with the Tram Car and Mountain Railway, but should show a cable and mountain, respectively, too
- 🚅🚃, 🚈🚃 – it doesn’t work well with any of the other left-facing trains, but should probably adapt its design accordingly
- 🚝🚃, 🚟🚃 – it’s worse with special types of rails
- 🚄🚃 - it’s worst with High-Speed Train because the glyph is not oriented like all the other ones (odd in the same way as Articulated Lorry 🚛 and Racing Car 🏎), also doesn’t really fit Station 🚉
- 🚆🚃, 🚇🚃, 🚊🚃 – it doesn’t work at all with front-facing Train, Metro and Tram, but probably also doesn’t have to
- 🛲🚃 – Diesel Locomotive doesn’t have a glyph in Emojione (because it’s not defined in Unicode as an emoji), but could have

If only either the left or the right glyph needs to change, this would use the `calt` Contextual Alternates feature in Opentype. If they combined into a single glyph, `clig` Contextual Ligature (or `liga`). If both (or multiple ones) needed to be substituted, I think it would only work by using a double-wide single glyph.

Other conventional combinations may be based on memes (🐸☕️), multi-part gestures (👉👌) or (English) compound words (🐮💩). Some proposals to Unicode list possible sequences for alternative representations and emojis get rejected because these alternatives are considered good enough for general use, e.g. some popular emoticons from [Emoji Request](http://emojirequest.com), Skype and Facebook in [L2/16-313](http://www.unicode.org/L2/L2016/16313-emoji-faces.pdf):

[![Excerpt from L2/16-314, annotated by me](http://pbs.twimg.com/media/Cw5Z5r8WEAAtUQ4.jpg:large)](http://www.unicode.org/L2/L2016/16314-emoji-faces-data.pdf)

If you’re interested in this at all, I could provide a larger list (which I’m still in the process of compiling).

**Comment by luckydonald** [comment]
PDF http://www.unicode.org/L2/L2016/16314-emoji-faces-data.pdf is 404.

**Comment by Crissov** [comment]
@luckydonald Indeed, there’s a typo in the file name on the server: the _i_ in _emoji_ is missing. I’ve updated the link accordingly.
